src/tablecellproperties/ui/tablecellpropertiesview.js

```diff
@@ -0,0 +1,707 @@
+			label: t( 'Cancel' ),
+			icon: cancelIcon,
+			class: 'ck-button-cancel',
+			type: 'cancel',
+			withText: true,
+		} );
+
+		cancelButtonView.delegate( 'execute' ).to( this, 'cancel' );
+
+		return {
+			saveButtonView, cancelButtonView
+		};
+	}
+
+	/**
+	 * Provides a set of {@link #borderStyleDropdown} item definitions.
+	 *
+	 * @private
+	 * @returns {Iterable.<module:ui/dropdown/utils~ListDropdownItemDefinition>}
+	 */
+	_getBorderStyleDefinitions() {
+		const itemDefinitions = new Collection();
+
+		for ( const style in this._borderStyleLabels ) {
+			const definition = {
+				type: 'button',
+				model: new Model( {
+					_borderStyleValue: style,
+					label: this._borderStyleLabels[ style ],
+					withText: true,
+				} )
+			};
+
+			definition.model.bind( 'isOn' ).to( this, 'borderStyle', value => {
+				return value === style;
+			} );
+
+			itemDefinitions.add( definition );
+		}
+
+		return itemDefinitions;
+	}
+
+	/**
+	 * Fills an alignment (either horizontal or vertical) with buttons
+	 * that have certain labels and interact with a certain view property
+	 * upon execution.
+	 *
+	 * @private
+	 * @param {module:ui/toolbar/toolbarview~ToolbarView} toolbar
+	 * @param {Array.<String>} labels
+	 * @param {String} propertyName
+	 */
+	_fillAlignmentToolbar( toolbar, labels, propertyName ) {
+		for ( const alignment in labels ) {
+			const button = new ButtonView( this.locale );
+
+			button.set( {
+				label: labels[ alignment ],
+				icon: ALIGNMENT_ICONS[ alignment ],
+			} );
+
+			button.bind( 'isOn' ).to( this, propertyName, value => {
+				return value === alignment;
+			} );
+
+			button.on( 'execute', () => {
+				this[ propertyName ] = alignment;
+			} );
+
+			toolbar.items.add( button );
+		}
+	}
+
+	/**
+	 * Provides localized labels for {@link #borderStyleDropdown} items.
+	 *
+	 * @private
+	 * @type {Object.<String,String>}
+	 */
+	get _borderStyleLabels() {
+		const t = this.t;
+
+		return {
+			none: t( 'None' ),
+			solid: t( 'Solid' ),
+			dotted: t( 'Dotted' ),
+			dashed: t( 'Dashed' ),
+			double: t( 'Double' ),
+			groove: t( 'Groove' ),
+			ridge: t( 'Ridge' ),
+			inset: t( 'Inset' ),
+			outset: t( 'Outset' ),
+		};
+	}
+
+	/**
+	 * Provides localized labels for {@link #horizontalAlignmentToolbar} buttons.
+	 *
+	 * @private
+	 * @type {Object.<String,String>}
+	 */
+	get _horizontalAlignmentLabels() {
+		const t = this.t;
+
+		return {
+			left: t( 'Align cell text to the left' ),
+			center: t( 'Align cell text to the center' ),
+			right: t( 'Align cell text to the right' ),
+			justify: t( 'Justify cell text' ),
+		};
+	}
+
+	/**
+	 * Provides localized labels for {@link #verticalAlignmentToolbar} buttons.
+	 *
+	 * @private
+	 * @type {Object.<String,String>}
+	 */
+	get _verticalAlignmentLabels() {
+		const t = this.t;
+
+		return {
+			top: t( 'Align cell text to the top' ),
+			middle: t( 'Align cell text to the middle' ),
+			bottom: t( 'Align cell text to the bottom' )
+		};
+	}
+}
```

Comment:
Just a thought - awfully long and mixes logic for different properties. I think that it might be a good candidate for a follow up / decision to make for the future. It's OK for the MVP :)

Reply:
ps.: Also the above comment can have little sense if we can't have a need for enabling or disabling some properties.